README.md

  
# Shaggle Core Payload CMS Plugin

A plugin for [Payload CMS](https://github.com/payloadcms/payload) to help speed up the development of websites for agencies with the goal of allowing installation within Vercel/Payload CMS Cloud hosting along with upcoming block library to keep all sites in sync with newest features and uniformity across agency sites.

## Important Note
`Version 1.x` is considered in active development; this release is considered part of the `Tabby` build and once we hit `Version 2.x` it'll be more battle tested and production ready.  While this goes against semantic versioning, this was an accident while initially being developed.

### Core features:

### Built In Standard Collections:

- Pages

- News (Blog)

- Media

- Rooms

- Attractions

- Golf Courses

- Rooms

- Properties

- Careers

### Built In Standard Globals:

- Developer Settings

- Global Settings

- Booking Engine

- Navigation

## Installation

```bash
yarn add  @kittabit/shaggle-core
# OR
npm i  @kittabit/shaggle-core
```

## Basic Usage

In the `plugins` array of your [Payload config](https://payloadcms.com/docs/configuration/overview), call the plugin with [options](#options):

```js
import { buildConfig } from  'payload/config';
import shaggleCore from  '@kittabit/shaggle-core';

const  config  =  buildConfig({
	plugins: [	
    	shaggleCore({
      		disableCollections: ['golf-courses', 'properties'],
    	}),
	]
});

export  default config;
```

### Options

#### `disableCollections`

By default, all collections are enabled; versus enabling one-by-one, this will allow you to simply disable the non-applicable items for your build/client via an array.

## Roadmap / Upcoming Features (Very Soon)

- User Updates (First Name, Last Name, Client Information, Roles, etc)
- Block Library Add-On (admin + frontend)
- Shaggle Website w/ Documentation, GraphQL Examples, Blocks, etc
- Demo Website (with admin + frontend)
- NextJS Base Build (with all features enabled)
- Blueshift / Revinate / OccasionGenius Plugins (separate repositories/plugins)

## Known Bugs
- FontAwesome Icon Missing (`features` Fields)
- Disable Settings for Non-Applicable/Activated Collections
- User Roles Missing
- Enable Preview's

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## [1.4.0] - 2023-04-25

### Added

### Changed

- REST API Access Control Updates
- `Navigation` Global Updates
- `GlobalSettings` Global Updates
- `BookingEngine` Global Updates

### Fixed

## [1.3.0] - 2023-04-24

### Added
- `DefaultSlider` Block
- `TwoColumnWithImage` Block
- `DefaultCTA` Block
- `ThreeColumnGrid` Block
- `DefaultFAQs` Block
- `content` Field to `News` Collection (versus only block layouts)
- `DefaultGallery` Block
- `DefaultContent` Block
- `Careers` Collection

### Changed
- Minor Collection Order Change

### Fixed
- `Attractions` updated `useAsTitle` for `title`

## [1.2.0] - 2023-04-22

### Added

- Ability to disable Collections from `payload.config.ts`

### Fixed

### Changed

## [1.1.0] - 2023-04-22

### Added

- `GolfCourses` Collection
- `Properties` Collection

### Fixed

- Missing `publishedDate,` `slug,` `author,` and `status` on multiple collections.

### Changed

- Navigation & Subnavigation Updates
- 'versions' Added to All Collections

### Removed

## [1.0.0] - 2023-04-12

### Added

- `Pages` Collection
- `News` Collection
- `Media` Collection
- `Rooms` Collection
- `Attractions` Collection
- `Booking Engine` Global
- `Developer Settings` Global
- `Global Settings` Global
- `Navigation` Global

### Fixed

### Changed

### Removed


src/collections/Properties.ts
import { CollectionConfig } from 'payload/types';

const Properties: CollectionConfig = {
	slug: 'properties',
    versions: true,
    admin: {
        useAsTitle: 'title',
    },
	fields: [
		{
			name: 'title',
			type: 'text',            
			required: true,
		},
        {
            type: 'tabs',
            tabs: [
                {
                    label: 'Basic', 
                    fields: [ 
                        {
                            name: 'headline',
                            label: "Headline",
                            type: 'text',
                            required: false,
                        },                  
                        {
                            name: 'intro',
                            label: "Intro Text/Description",
                            type: 'textarea',
                            required: false,
                        },  
                    ],
                },
                {
                    label: 'Detailed', 
                    fields: [ 
                        {
                            name: 'description',
                            label: "Description",
                            type: 'richText',
                            required: false,
                        },
                        {
                            name: 'features',
                            label: "Property Features",
                            minRows: 0,
                            maxRows: 24,            
                            type: 'array',
                            labels: {
                                singular: 'Feature',
                                plural: 'Features',
                            },
                            fields: [
                                {
                                    name: 'icon',
                                    type: 'text',
                                },
                                {
                                    name: 'label',
                                    type: 'text',
                                },
                                {
                                    name: 'featured',
                                    label: "Featured Feature",
                                    type: 'checkbox',
                                    defaultValue: false,
                                },                
                            ],
                        }, 
                    ],
                },  
                {
                    label: 'Location', 
                    fields: [ 
                        {
                            type: 'row',
                            fields: [ 
                                {
                                    name: 'streetAddress1',
                                    label: "Street Address",
                                    type: 'text',
                                    admin: {
                                        width: "50%",
                                    },
                                  },
                                  {
                                    name: 'streetAddress2',
                                    label: "Street Address 2",
                                    type: 'text',
                                    admin: {
                                        width: "50%",
                                    },                
                                  },              
                                  {
                                    name: 'city',
                                    label: "City",
                                    type: 'text',
                                    admin: {
                                        width: "50%",
                                    },                                    
                                  },           
                                  {
                                    name: 'state',
                                    label: "State",
                                    type: 'text',
                                    admin: {
                                        width: "25%",
                                    },                                    
                                  },                 
                                  {
                                    name: 'postalCode',
                                    label: "Postal Code",
                                    type: 'text',
                                    admin: {
                                        width: "25%",
                                    },                                    
                                  },  
                            ],
                        },
                        {
                            name: 'contactInformation',
                            label: "Contact Information",
                            type: 'group',
                            fields: [
                                {
                                    type: 'row',
                                    fields: [
                                        {
                                            name: 'phoneNumber',
                                            label: "Phone Number",
                                            type: 'text',
                                            required: false,
                                            admin: {
                                                width: "50%",
                                            },                                     
                                        },                  
                                        {
                                            name: 'websiteUrl',
                                            label: "Website URL",
                                            type: 'text',
                                            required: false,
                                            admin: {
                                                width: "50%",
                                            },                                     
                                        },          
                                    ],
                                },
                            ],
                        },
                        {
                            name: 'geoSpatialInformation',
                            label: "Geospatial",
                            type: 'group',
                            fields: [ 
                                {
                                    type: 'row',
                                    fields: [
                                        {
                                            name: 'latitude',
                                            type: 'text',
                                        },
                                        {
                                            name: 'longitude',
                                            type: 'text',
                                        },    
                                    ],
                                }          
                            ],
                        },
                    ],
                },    
                {
                    label: 'Photography', 
                    fields: [ 
                        {
                            name: 'propertyLogo',
                            label: "Property Logo",
                            type: 'upload',
                            relationTo: 'media',
                            required: false,
                        }, 
                        {
                            name: 'smallThumbnail',
                            label: "Small Thumbail",
                            type: 'upload',
                            relationTo: 'media',
                            required: false,
                        },                                  
                        {
                            name: 'largeThumbnail',
                            label: "Large Thumbail",
                            type: 'upload',
                            relationTo: 'media',
                            required: false,
                        },                  
                        {
                            name: 'imageGallery',
                            label: "Image Gallery",
                            minRows: 0,
                            maxRows: 50,            
                            type: 'array',
                            labels: {
                                singular: 'Image',
                                plural: 'Images',
                            },
                            fields: [
                                {
                                    name: 'altText',
                                    type: 'text',
                                },
                                {
                                    name: 'galleryImage',
                                    type: 'upload',
                                    relationTo: 'media',
                                    required: false,
                                },
                            ],
                        }, 
                    ],
                }, 
                {
                    label: 'Booking Engine', 
                    fields: [ 
                        {
                            type: 'row',
                            fields: [
                                {
                                    name: 'bookingUrl',
                                    label: "Booking URL",
                                    type: 'text',
                                    required: false,
                                    admin: {
                                        width: "75%",
                                    },                                     
                                },                  
                                {
                                    name: 'bookingId',
                                    label: "Booking ID",
                                    type: 'text',
                                    required: false,
                                    admin: {
                                        width: "25%",
                                    },                                     
                                },          
                            ],
                        }
                    ],
                },
            ],
        },
        {
			name: 'slug',
			type: 'text',
			required: true,
			admin: {
				position: 'sidebar',
			},			
		},		
		{
			name: 'status',
			type: 'select',
			options: [
				{
					value: 'draft',
					label: 'Draft',
			  	},
			  	{
					value: 'private',
					label: 'Private',
			  	},				
			  	{
					value: 'published',
					label: 'Published',
			  	},
			],
			defaultValue: 'draft',
			admin: {
				position: 'sidebar',
			}
		},
		{
			name: 'publishedDate',
			type: 'date',
			admin: {
				position: 'sidebar',
			}
		},
		{
			name: 'author',
			type: 'relationship',
			relationTo: 'users',
			admin: {
			  position: 'sidebar',
			}
		},  
	]
};

export default Properties;